### PR TITLE
Improve cancelled builds treatment

### LINKED ISSUE #31: Stop treating "canceled" builds as "failed" ones
We should reconsider how we treat canceled builds while the cancellation is an intentional action from one of the core developers.

Let's have a [different icon](https://icon-icons.com/icon/stop-circle-outline/139164) for the canceled builds and not treat them as failed ones anymore.

### PR DESCRIPTION
**Pull request by yuri-sergiichuk**
This PR closes #31.

From now on, we're going to send a different message for canceled builds. Also, canceled builds are going to be treated as "stable" if there are 2 or more consecutive cancellations.

### CODE REVIEW

File: google-chat-bot/src/main/java/io/spine/chatbot/server/google/chat/ThreadChatRepository.java [hunk 1]
```diff
@@ -63,7 +64,8 @@
     protected void setupEventRouting(EventRouting<ThreadId> routing) {
         super.setupEventRouting(routing);
         routing.route(BuildFailed.class, new RepositoryEventRoute<>())
-               .route(BuildRecovered.class, new RepositoryEventRoute<>());
+               .route(BuildRecovered.class, new RepositoryEventRoute<>())
```
Comment (dmdashenkov) [inline google-chat-bot/src/main/java/io/spine/chatbot/server/google/chat/ThreadChatRepository.java:67]: Here, "Repository" of Git clashes with "Repository" of Spine. Please rename this `EventRoute` class to make it more telling. My suggestion is `ByGitRepository`.

Comment (dmdashenkov) [inline google-chat-bot/src/main/java/io/spine/chatbot/server/google/chat/ThreadChatRepository.java:67]: Why do we need a class here at all? Could that be a single method of the `ThreadChatRepository` with a method reference here?

Comment (dmdashenkov) [reply google-chat-bot/src/main/java/io/spine/chatbot/server/google/chat/ThreadChatRepository.java:67]: Is it something to do with the Java serialization?

**Review by dmdashenkov — COMMENTED** [review]
LGTM in general. Please see my comments and questions.

File: google-chat-bot/src/main/java/io/spine/chatbot/server/google/chat/ThreadChatRepository.java (hunk 1 above)
Comment (yuri-sergiichuk) [reply google-chat-bot/src/main/java/io/spine/chatbot/server/google/chat/ThreadChatRepository.java]: Replaced with a method reference.

**Comment by yuri-sergiichuk** [comment]
@dmdashenkov PTAL again.

**Review by dmdashenkov — APPROVED** [review]

**Pull request merged**